### ISSUE TITLE
Upload speed issues

### ISSUE THREAD

**Issue by meduselchen** [becast/YouPloader#18]
I have 10 Mbit/s upload, but with Youploader I can only get max ~300 KB/s each upload:
http://i.imgur.com/dkseflz.jpg

**Comment by becast** [comment]
This is a bug that has been fixed here. Will be fixed with 0.6. Sorry about that. If you want to fix it in the meantime download this file: https://build.becast.at/job/YouPloader/112/artifact/target/YouPloader.jar and place it in your YouPloader dir.

**Comment by becast** [comment]
Got a report from elsewhere that this might still is an issue with 0.6. Can you confirm?

**Comment by ClefAria** [comment]
I believe I'm the 'report from elsewhere' you're referring to. I tried using the linked .jar here and still have the same issue.

Is the fact that I'm using an SSD relevant, maybe? Since I saw that the attempted fix from before had to do with disk access.

**Comment by becast** [comment]
Can you please attach your logfile.html? Just go to "?"-> "View Logfile"

**Comment by ClefAria** [comment]
Github wouldn't accept the .html file, so I sipped it up.
[logfile.zip](https://github.com/becast/YouPloader/files/264158/logfile.zip)

**Comment by becast** [comment]
The lates Release is here: https://github.com/becast/YouPloader/releases/tag/v0.6-alpha will check out your issue later today. Sorry that you experience issues. It's probably unrelated to your SSD as it would probably mitigate the issue if it was due to disk access :)

**Comment by ClefAria** [comment]
I didn't know this program existed before 0.6. I only reverted to the .jar in this thread today, to experiement. I've already gone back to 0.6. Should I delete the logfile and let it start a new one from scratch?

**Comment by becast** [comment]
It's okay just include the logfile the output is versioned

**Comment by ClefAria** [comment]
The logfile is included in my previous comment.

**Comment by becast** [comment]
I see nothing obvious in the second logfile. The first one has some NPEs. What Java Version do you have installed?

**Comment by meduselchen** [comment]
sorry for the late answer, here I am again :)

yes I still have the same problem with 0.6 alpha. 
log: http://pastebin.com/kYz3e4W5
also made the debug log for you: http://pastebin.com/9wXqbudh

running on: Java 8.0.910.15 | Win 7 64x

**Comment by becast** [comment]
Thanks, this is absolutely weird and looking like some random limitation, I googled and found a similar issue with no solution. Are you running 64bit Java?

**Comment by becast** [comment]
I made a little change, adding more debug code and trying something in a stab in the dark approach.

Can you download this and replace your Youploader.exe and your Youploader.jar with those?
[Issue18 Test.zip](https://github.com/becast/YouPloader/files/264997/Issue18.Test.zip)

**Also the debug command now needs  a "-" in front, so it needs to be -debug**

**Comment by ClefAria** [comment]
I've got 8 GB of RAM.

Tried the test files, and the upload speed improved, but only a little. I was averaging about 850 KB/s.

This is what I have re Java: http://i.imgur.com/hHiWNa6.png

**Comment by becast** [comment]
Well 850kb/s vs. 300kb/s is more than 50% improvement xD So at least we're working into the right direction. Could you please also attach a debug log as described above? Thank you!

**Comment by ClefAria** [comment]
I made two attempts with -debug on. During the first, I forgot I was downloading something, so I thought it might have been affecting the upload, so I finished it and then tried again, but I got the same result, upload hitting a ceiling of about 850 KB/s
[logfile.zip](https://github.com/becast/YouPloader/files/265042/logfile.zip)

**Comment by becast** [comment]
Okay since the Buffer helped somewhat I upped the buffer size to 5MB for this test. Can you test if this does improve the speed?
[Issue18 Test 2.zip](https://github.com/becast/YouPloader/files/265046/Issue18.Test.2.zip)

**Comment by ClefAria** [comment]
No improvement, that I can see. Sometimes, YouTube seems to limit its uploading speed, and during those times, uploading via the website caps out at 2.5 MB/s instead of my max of 3.5 MB/s...thought I'd mention this just in case there's some way it might be related. Did a test upload on the site and am getting 2.5 right now.

Can't break 900 in the program, though. :/
[logfile.zip](https://github.com/becast/YouPloader/files/265053/logfile.zip)

**Comment by becast** [comment]
Weird, I fear this might stem from the Apache HttpClient library YouPloader uses and the fact that my Internet has no hopes of ever reaching those speeds makes debugging hard for me. I'll try my best to figure out a solution, but I cannot offer a time frame. Maybe someone with faster internet can take a look at it. Sorry :( For now I have reached the limit of what I can do.

**Comment by meduselchen** [comment]
Got 16Gb RAM, HDD write speed: 121MB/s and Java x64:
_java version "1.8.0_91"
Java(TM) SE Runtime Environment (build 1.8.0_91-b14)
Java HotSpot(TM) 64-Bit Server VM (build 25.91-b14, mixed mode)_


No improvements for me so far (can't upload zip-files...):
_Issue18 Test_: https://www.odrive.com/s/2ac29ef0-27d2-4e6d-b1da-bfd506eb26e5-5738a0ea
_Issue18 Test2_: https://www.odrive.com/s/9e46dd62-e085-435c-84d3-250003826f84-5738a10e

**Comment by becast** [comment]
Yours is definitely the weirdest since it is way slower for whatever reason. I just saw that you both use Java 1.0.8_91... I'm going to test something.

**Comment by ClefAria** [comment]
I had a weird combination of things, trying to find the thing that would prevent this program from scolding me when I started it up.

I've since uninstalled them all, so if you have one to recommend for another test, I'm game.

**Comment by becast** [comment]
I'm sorry I don't follow. What do you mean with " trying to find the thing that would prevent this program from scolding me when I started it up"

**Comment by ClefAria** [comment]
When I first downloaded the program, it complained that I didn't have (if I remember correct) "Oracle Java 1.8 or newer" installed. It looks like Java's version numbering system doesn't work like that, so I ended up trying a few things, and eventually, it stopped complaining when I opened it. I took a screenshots of the three Java 'things' I had installed during all this.

**Comment by becast** [comment]
Okay, the installer complaining is also weird but most likely unrelated. I just did a test with 1.8.0_91 (YouPloader is developed with 1.8.0_92) and it still works. I was just grasping at a straw that the small update might have been related. But not closer to figuring out why your YouPloader decides to drive with its handbrake on.

**Comment by ClefAria** [comment]
What about x86 vs. x64? Which did you use? Could it matter?

**Comment by becast** [comment]
I'm using x64, same as @meduselchen. Theoretically it should not matter.

**Comment by ClefAria** [comment]
Just started from scratch, with the portable version (used installer before), replaced with test 2 .exe and .jar, installed java v. 8 update 92 x64, and I'm back at about ~500 KB/s uploading.

**Comment by becast** [comment]
Do both of you use a firewall or Internet Security package? Could you disable it for a test? It would also help if you had a different computer that you could test on your connection.

**Comment by Filgaja** [comment]
i am Using Windows 10 x64 and the latest java jre1.8.0_112 and there is no issue for me :)

I have 100/10 Internet Connection and i upload around with 1 Mbit/s

Maybe its also a router problem? do you have costum/standart ports they maybe have to be aneabled? Also a lot of Router have an firewall included.

Shall we open a port for YouPLoader?

**Comment by becast** [comment]
I believe this is fixed since 0.9.1

**Comment by ClefAria** [comment]
You would be incorrect. Just redownloaded and installed 0.9.1, still locked at 850 KB/s. Uninstalling again. :/

**Comment by Donzii** [comment]
Cannot confirm, uploads here with fullspeed

**Comment by ClefAria** [comment]
What is "full speed" for you, Donzii? For me, it's about 3.5 MB/s.

**Comment by Donzii** [comment]
DL: 53,5 Mbit/s
UL: 10,7 Mbit/s

Fullspeed = ~1000kb/s

**Comment by ClefAria** [comment]
Well, guess I'll have to wait for confirmation from someone who can crack at least 2 MB/s up. It's looking like I have the highest max upload speed of anyone who's tried the program, dev included.

**Comment by becast** [comment]
Dumb question, but are you reaching those speeds with the Youtube website when you upload?

**Comment by ClefAria** [comment]
Yes, occasionally it randomly caps at 2.5 MB/s instead, but that happened much more frequently about a year ago. Now, I'm rarely not over 3 MB/s upload on YouTube.

I'm on Win7 and have UAC completely disabled, everything I run is 'as admin', but I guess I could try if I reinstalled again, when I get a chance.

**Comment by ClefAria** [comment]
Reinstalled, ran as administrator. This time, I capped out at about 600 KB/s. Cancelled and immediately fired up YouTube in the browser, went to upload the same file, and instantly got ~3.3 MB/s up. Don't know what to tell you.

**Comment by meduselchen** [comment]
yeah, still the same problem on my end as well.
v. 0.9.: max 200KB/s
right now via browser: ~9 Mbit/s

don't know whats happening here.

**Comment by Zero3K** [comment]
My upload speed is fluctuating between 300-565 KB/s (which is the max speed of my Internet Plan). Maybe its because of the Wireless Network I am using (which is a dongle plugged into my PC that is providing WiFi by using its Ethernet Connection).
